gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/InitialImageOperation.java

```diff
@@ -2795,6 +2797,20 @@ final public int getProcessorType() {
       return this.targetReinitialized ? DistributionManager.WAITING_POOL_EXECUTOR :
                                 DistributionManager.HIGH_PRIORITY_EXECUTOR;
     }
+
+    /**
+     * GII operation will wait if there are some running transaction after
+     * StateFlush message. This method will make GII wait till all those transactions
+     * are finished.
+     */
+    protected void waitForRunningTXs(DistributedRegion rgn,
+                                     InternalDistributedMember sender) {
+     // Wait for all write operations to get over.
+     BucketRegion bucketRegion = (BucketRegion)rgn;
+      InitializingBucketMembershipObserver listener =
```

Comment:
We are handling only cases where the GII requestor dies down which will be a rare situation. I don't think we should be keeping a listener only for that. Hence cleaning up the listener once GII is either done or requesting member dies.